### PR TITLE
Switched to dolly from crate to a submodule dependency

### PR DESCRIPTION
**Pull request by BlackPhlox**
Based on the conversation in https://github.com/BlackPhlox/bevy_dolly/pull/25#discussion_r1125426264, I've gone ahead and added [dolly as a submodule](https://github.com/BlackPhlox/dolly/tree/bevy_dolly) and remove its own dependencies and transform and replaced them with bevy's implementation.

### CODE REVIEW

**Comment by BlackPhlox** [comment]
Related to PR https://github.com/BlackPhlox/bevy_dolly/pull/25

**Comment by BlackPhlox** [comment]
The CI passes now, ready for review!

**Comment by janhohenheim** [comment]
Thanks, I'll check in a few hours.
Sorry for taking a while, was busy moving 😄

**Comment by BlackPhlox** [comment]
No worries!

File: examples/2d_cam_move.rs [hunk 1]
```diff
@@ -32,9 +32,10 @@ fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
             .with(Position::new(Vec3::new(0., 0., 400.)))
             .with(Smooth::new_position(1.2))
             .build(),
-        Camera2dBundle::default(),
     ));
 
+    commands.spawn(Camera2dBundle::default()).insert(MainCamera);
```
Comment (janhohenheim) [inline examples/2d_cam_move.rs:37]:
```suggestion
    commands.spawn((Camera2dBundle::default(), MainCamera));
```

**Pull request merged**